benchmarking/compute_probability_of_improvement.py

```diff
@@ -0,0 +1,251 @@
+            f"samples will be truncated to the minimum sample count of"
+            f" {min_sample_count}",
+        )
+
+    return np.asarray(
+        [
+            [
+                run["result"]["imit_stats"]["monitor_return_mean"]
+                for run in runs_by_env[env][:min_sample_count]
+            ]
+            for env in envs
+        ],
+    ).T
+
+
+def compute_probability_of_improvement(
+    runs_by_env: Dict[str, List[SacredRun]],
+    baseline_runs_by_env: Dict[str, List[SacredRun]],
+    reps: int,
+):
+    """Computes the probability of improvement of the runs over the baseline runs.
+
+    Args:
+        runs_by_env: A dictionary mapping environment names to lists of runs.
+        baseline_runs_by_env: A dictionary mapping environment names to lists of runs.
+        reps: The number of bootstrap repetitions to use to compute the confidence
+            interval.
+
+    Returns:
+        A tuple of:
+         - probability of improvement
+         - confidence interval
+         - number of samples per env
+         - number of baseline samples per env
+    """
+    envs = runs_by_env.keys()
+    baseline_envs = baseline_runs_by_env.keys()
+    comparison_envs = sorted(set(envs).intersection(set(baseline_envs)))
+
+    run_scores = sample_matrix_from_runs_by_env(runs_by_env, comparison_envs)
+    baseline_run_scores = sample_matrix_from_runs_by_env(
+        baseline_runs_by_env,
+        comparison_envs,
+    )
+    samples_per_env = run_scores.shape[0]
+    baseline_samples_per_env = baseline_run_scores.shape[0]
+
+    probabs, error_intervals = rly.get_interval_estimates(
+        {"baseline_vs_new": (baseline_run_scores, run_scores)},
+        metrics.probability_of_improvement,
+        reps=reps,
+    )
+    probability_of_improvement = probabs["baseline_vs_new"]
+    confidence_interval = np.squeeze(error_intervals["baseline_vs_new"])
+
+    return (
+        probability_of_improvement,
+        confidence_interval,
+        samples_per_env,
+        baseline_samples_per_env,
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("runs_dir", type=pathlib.Path)
+    parser.add_argument("baseline_runs_dir", nargs="?", default=None, type=pathlib.Path)
+    parser.add_argument("--baseline-algo", type=str)
+    parser.add_argument("--algo", type=str)
+    parser.add_argument("--bootstrap-reps", type=int, default=2000)
+
+    args = parser.parse_args()
+
+    if args.baseline_runs_dir is None:
+        args.baseline_runs_dir = args.runs_dir
+
+    runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.runs_dir,
+        only_completed_runs=True,
+    )
+    baseline_runs_by_algo_and_env = group_runs_by_algo_and_env(
+        args.baseline_runs_dir,
+        only_completed_runs=True,
+    )
+
+    algos = sorted(runs_by_algo_and_env.keys())
+    baseline_algos = sorted(baseline_runs_by_algo_and_env.keys())
```

Comment:
I think it would be more clear to call these `algos_in_runs_dir` and `algos_in_baselines_dir` to avoid clashes with the attributes of `args`.

Reply:
Very good point. Changed that.